Implement forced-colors media query.

mdubet opened the pull request:
#### 46513cc172ce079baedb34e890803058b9158b35
<pre>
Implement forced-colors media query.
<a href="https://bugs.webkit.org/show_bug.cgi?id=225281">https://bugs.webkit.org/show_bug.cgi?id=225281</a>
rdar://77715265

Reviewed by Tim Nguyen and Aditya Keerthi.

MQ always matchs &quot;none&quot;, as there is no concept of &quot;forced colors&quot; in
Cocoa.
<a href="https://www.w3.org/TR/mediaqueries-5/#forced-colors">https://www.w3.org/TR/mediaqueries-5/#forced-colors</a>

* LayoutTests/imported/w3c/web-platform-tests/css/mediaqueries/forced-colors-expected.txt:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/MediaFeatureNames.h:
* Source/WebCore/css/MediaQueryEvaluator.cpp:
(WebCore::forcedColorsEvaluate):
* Source/WebCore/css/MediaQueryExpression.cpp:
(WebCore::isValidValueForIdentMediaFeature):
(WebCore::featureWithValidIdent):
(WebCore::isFeatureValidWithoutValue):

Canonical link: <a href="https://commits.webkit.org/253290@main">https://commits.webkit.org/253290@main</a>
</pre>

pxlcoder commented on Source/WebCore/css/MediaQueryEvaluator.cpp:
```diff
@@ -489,6 +489,17 @@ static bool scanEvaluate(CSSValue* value, const CSSToLengthConversionData&, Fram
+    if (!primitiveValue)
+        return false;
+
+    // On macOS, we don't support "High Contrast" mode,
```
`On macOS` -> `On Cocoa platforms`

Also, there is a "High Contrast" mode on macOS and iOS (using the "Increase Contrast" setting), but there is no concept of "forced colors".

mdubet replied:
I will reword the comment, it's not very accurate indeed.

For reference, WK already supports the "increase contrast setting" via the "prefers-contrast" MQ : https://github.com/WebKit/WebKit/blob/e66751caa5cc6a8968a262df6c027f19dcc6c0f2/Source/WebCore/css/MediaQueryEvaluator.cpp#L840

pxlcoder commented on Source/WebCore/css/MediaQueryEvaluator.cpp:
```diff
@@ -489,6 +489,16 @@ static bool scanEvaluate(CSSValue* value, const CSSToLengthConversionData&, Fram
+    if (!primitiveValue)
+        return false;
+
+    // On Cocoa platform, there is no concept of "forced colors".
```
Nit: `platform` -> `platforms`.

mdubet committed:
Implement forced-colors media query.
https://bugs.webkit.org/show_bug.cgi?id=225281
rdar://77715265

Reviewed by Tim Nguyen and Aditya Keerthi.

MQ always matchs "none", as there is no concept of "forced colors" in
Cocoa.
https://www.w3.org/TR/mediaqueries-5/#forced-colors

* LayoutTests/imported/w3c/web-platform-tests/css/mediaqueries/forced-colors-expected.txt:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/MediaFeatureNames.h:
* Source/WebCore/css/MediaQueryEvaluator.cpp:
(WebCore::forcedColorsEvaluate):
* Source/WebCore/css/MediaQueryExpression.cpp:
(WebCore::isValidValueForIdentMediaFeature):
(WebCore::featureWithValidIdent):
(WebCore::isFeatureValidWithoutValue):

Canonical link: https://commits.webkit.org/253290@main
